src/components/security_manager/src/crypto_manager_impl.cc

```diff
@@ -235,7 +243,15 @@ bool CryptoManagerImpl::OnCertificateUpdated(const std::string& data) {
     return false;
   }
 
-  return set_certificate(data);
+  if (!set_certificate(data)) {
+    LOG4CXX_ERROR(logger_, "Failed to save certificate data");
+    return false;
+  }
+
+  X509* module_certificate = LoadModuleCertificateFromFile();
+  EVP_PKEY* module_key = LoadModulePrivateKeyFromFile();
```

Comment:
Shouldn't we free allocated resources (X509 and EVP_PKEY)?

Reply:
+1 I see BIO_free, but didn't see X509_free & EVP_PKEY_free

Reply:
@KVGrygoriev Maybe I am wrong but I thought that BIO_free cleans up it as well.

Reply:
@olohvinenko @KVGrygoriev @AByzhynar 
Thank you guys, good point. I added scope guards for all allocated X509 and EVP_PKEY objects so the will be automatically released when guard will comes out of scope.
See 65a8cb2fce55a81ed184b172d4c9490ea00e1f3a